observation_portal/requestgroups/pattern_expansion.py

```diff
@@ -38,10 +31,61 @@ def expand_dither_pattern(dither_details):
     return configuration_dict
 
 
+def expand_mosaic_pattern(expansion_details):
+    '''
+    Takes in a valid request with one configuration and valid set of mosaic parameters,
+    and expands the configuration within the request with new targets using offsets to fit the
+    mosaic pattern details specified.
+    :param expansion_details: a valid dictionary containing the `request`, and a number of pattern expansion
+                           parameters such as `num_points`, `pattern`, `point_spacing` and others.
+    :return: Request with expanded list of configurations with different targets following pattern.
+    '''
+    request_dict = expansion_details.get('request', {})
+    offsets = []
+    configurations = []
+    configuration = request_dict.get('configurations', [{}])[0]
+
+    offsets = expand_pattern(expansion_details)
+
+    for offset in offsets:
+        configuration_copy = deepcopy(configuration)
+        configuration_copy['target']['dec'] += (offset[1] / 3600.0)
+        cos_dec = cos(radians(configuration_copy['target']['dec']))
+        cos_dec = max(cos_dec, 10e-4)
+        configuration_copy['target']['ra'] += (offset[0] / 3600.0 / cos_dec)
+        extra_params = configuration_copy.get('extra_params', {})
+        extra_params['obs_recipe'] = 'Mosaic'
```

Comment:
Not sure if this would make more sense at the request level extra_params. Unless we end up having requests where some configurations are part of the mosaic and some aren't

Reply:
hmm yeah im not sure, its technically possible. I wonder if we should brand obs_recipe as more of a pass-through, so it can be used for mosaic in this case but other things could be stored in there (cadence perhaps?). So I guess it would make sense in the Request for mosaics/cadences... or if it included any dither info it would need to be in the configuration.

Reply:
oh silly me, there isn't a request level extra_params right now...

Reply:
Oh right, for some reason I also thought we had an extra params at that level... Ok well thinking about it more, the idea of potentially having a passthrough for something like this sounds reasonable, but I guess then we wouldn't be able to set when something is both a mosaic and a cadence, it could get messy just using one field, and I'm not sure we even need to go that far.... maybe, but the use cases aren't obvious to me at this moment. Let's just stick with this and if need arises for passing though more general observing mode/ observing recipes/ whatever, then we can revisit.